torchtext/csrc/vectors.cpp

```diff
@@ -0,0 +1,59 @@
+#include <string>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct Vectors : torch::CustomClassHolder {
+private:
+  std::unordered_map<std::string, int> stoi_;
+
+public:
+  // tokens_ and vectors_ holds the serialized params passed in during
+  // initialization. We need this because we need to be able to serialize the
+  // model so that we can save the scripted object. Pickle will get the
+  // serialized model from these members, thus they needs to be public.
+  std::vector<std::string> tokens_;
+  std::vector<torch::Tensor> vectors_;
+
+  explicit Vectors(std::vector<std::string> tokens,
+                   std::vector<torch::Tensor> vectors)
+      : tokens_(tokens), vectors_(vectors) {
+    for (std::size_t i = 0; i < tokens.size(); i++) {
+      stoi_[tokens[i]] = static_cast<int>(i);
```

Comment:
Why don't we save the `vectors` directly in `stoi_`, like `std::unordered_map<std::string, std::vector<torch::Tensor>>`

Reply:
Good call. I'll go ahead and do that. I think we still need to store the vectors in `vectors_` for the `__getstate__` and `__setstate__` methods.